src/Kiota.Builder/Writers/TypeScriptWriter.cs

```diff
@@ -175,22 +175,45 @@ public override void WriteMethod(CodeMethod code)
             WriteLine($"{GetAccessModifier(code.Access)} readonly {code.Name.ToFirstCharacterLowerCase()} = {(code.IsAsync ? "async ": string.Empty)}({string.Join(", ", code.Parameters.Select(p=> GetParameterSignature(p)).ToList())}) : {(code.IsAsync ? "Promise<": string.Empty)}{GetTypeString(code.ReturnType)}{(code.ReturnType.IsNullable ? " | undefined" : string.Empty)}{(code.IsAsync ? ">": string.Empty)} => {{");
             IncreaseIndent();
             var returnType = GetTypeString(code.ReturnType);
+            var requestBodyParam = code.Parameters.FirstOrDefault(x => x.ParameterKind == CodeParameterKind.RequestBody);
+            var queryStringParam = code.Parameters.FirstOrDefault(x => x.ParameterKind == CodeParameterKind.QueryParameter);
+            var headersParam = code.Parameters.FirstOrDefault(x => x.ParameterKind == CodeParameterKind.Headers);
```

Comment:
These three lines repeat in all the writers. Consider wrapping the whole thing with a tuple returning function.

Reply:
that's a dilemma I had: should the writers share any code between one another? I was leaning towards "no" because I don't want changes made for Java to impact the dotnet experience for example, but I'm curious to hear your thoughts on this?

Reply:
I understand and agree with the concern. If we start adding logic into a shared place, sometime in the future, we may start seeing people adding things like `if (language == Java)`, which is not ideal at all. I am retracting my comment on combining the 3 lines as a single function, but maybe the `code` object can have `ParametersOfKind()` method so that we don't have to repeat the filters.

Reply:
yeah I added the extension method on the IEnumerable<CodeParameters> to avoid having to type FirstOrDefault(x => x.ParameterKind.... all the time https://github.com/microsoft/kiota/pull/48/commits/08beb5d8e6f85df03433c6acabf61143baedeaf7

Unless you meant something else?

Reply:
That's what I meant, thanks!